File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    if(s[0] === ')') return false;
    let temp = 0
  
    for(let i=0; i<s.length; i++){
      if(s[i]==='('){
          temp++;
      }else{
          temp--;
      }
      if(temp < 0) return false;
    }
    return temp ? false : true
}

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        int answer = 0;
        
        for(char ch : s.toCharArray()){
            if(ch == '('){
                answer++;
            }else{
                answer--;
            }
            if(answer < 0) return false;
        }
        return answer == 0 ? true : false;
    }
}